src/query-store.ts

```diff
@@ -333,9 +333,17 @@ export const useQueryCache = defineStore(QUERY_STORE_ID, ({ action }) => {
    * @param options.refetch - if true, it will refetch the data
    */
   const invalidate = action((entry: UseQueryEntry) => {
-    // will force a fetch next time
-    entry.when = 0
+    // TODO: once `fetch` cancels query, move `cancelQuery` to the second condition
     cancelQuery(entry)
+    if (entry.deps.size > 0) {
+      // reset any pending request
+      entry.pending = null
+      // force refresh
+      fetch(entry)
```

Comment:
I don't think the invalidate should fetch. I think it's better to add a different action that fetches only active queries or maybe introduce a readonly property on the entry `active` or similar that can be checked on the mutation side

Reply:
That's a good point. Since I haven't found other use cases for it (checking if a query is active and refetching it), I moved it on the mutation side.